packages/coding-components/templates/angular/src/app/app.module.ts

```diff
@@ -13,20 +13,35 @@
  * limitations under the License.
  */
 
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
-  // Must be included for component packages that don't yet have an Angular wrapper
-import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { APP_INITIALIZER, NgModule } from "@angular/core";
+import { BrowserModule } from "@angular/platform-browser";
+// Must be included for component packages that don't yet have an Angular wrapper
+import { CUSTOM_ELEMENTS_SCHEMA } from "@angular/core";
+import { AppComponent } from "./app.component";
+
 // Angular wrapper for coding components
-import { ComponentLibraryModule } from '@arcgis/coding-components-angular';
+import { ComponentLibraryModule } from "@arcgis/coding-components-angular";
+import { defineCustomElements as defineCalciteElements } from "@esri/calcite-components/dist/loader";
+import { defineCustomElements as defineCodingElements } from "@arcgis/coding-components/dist/loader";
 
-import { AppComponent } from './app.component';
+const initializeCustomElements = () => {
+  // define custom elements in the browser, and load the assets from the CDN
+  defineCalciteElements(window, { resourcesUrl: "https://js.arcgis.com/calcite-components/2.4.0/assets" });
+  defineCodingElements(window, { resourcesUrl: "https://js.arcgis.com/coding-components/next/assets" });
+};
 
 @NgModule({
   declarations: [AppComponent],
   imports: [BrowserModule, ComponentLibraryModule],
   schemas: [CUSTOM_ELEMENTS_SCHEMA],
-  providers: [],
-  bootstrap: [AppComponent],
+  providers: [
+    {
+      provide: APP_INITIALIZER,
+      useFactory: () => initializeCustomElements,
```

Comment:
Just a heads-up, this is likely to trip up customers who don't understand factory providers or can't use this approach due to their existing application architecture. Often when that happens they'll call support or say it's a bug or defect in the SDK. Is there a more generic way to initialize the component library?

Reply:
unfortunately there is not. The only alternative would be to include `schema: [CUSTOM_ELEMENTS_SCHEMA]`, and removing the wrappers from the root `NgModule`. FWIW I spent a lot of time on this Monday and came to the conclusion that this is the cleanest and most angular compliant approach. I'll send you a doc on the subject.

Reply:
I chatted with Nick about investigating using `ngOnit()` which is a standard Angular component life-cycle event like mapViewReady.